linked_list/implementation.py

```diff
@@ -1,40 +1,135 @@
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
-
     def __init__(self, elements=None):
-        pass
-
+        if not elements:
+            self.start = None
+            self.end = None
+        else:
+            new_node = Node(elements[0])
+            self.start = new_node
+            self.end = new_node
+            for element in elements[1:]:
+                self.append(element)
+        
     def __str__(self):
-        pass
+        l = []
+        iterNode = self.start
+        while iterNode is not None:
+            l.append(iterNode.elem)
+            iterNode = iterNode.next
+        return str(l)
+        
 
     def __len__(self):
-        pass
+        iterNode = self.start
+        count = 0
+        while iterNode is not None:
+            count += 1
+            iterNode = iterNode.next
+        return count
 
+# http://stackoverflow.com/questions/2180578/python-why-cant-i-iterate-over-a-list-is-my-exception-class-borked
     def __iter__(self):
-        pass
+        self.iterNode = self.start
+        # return iter(self)
+        return self
+
+    def __next__(self):
+        if self.iterNode == None:
+            raise StopIteration()
+        value = self.iterNode.elem
+        self.iterNode = self.iterNode.next
+        return value
+        
+    def next(self):
+        if self.iterNode == None:
+            raise StopIteration()
+        value = self.iterNode.elem
+        self.iterNode = self.iterNode.next
+        return value
 
     def __getitem__(self, index):
-        pass
+        iterNode = self.start
+        count = 0
+        while count < index:
+            count += 1
+            iterNode = iterNode.next
+        return iterNode.elem
 
     def __add__(self, other):
-        pass
+        for item in other:
+            self.append(item)
+        return self
 
     def __iadd__(self, other):
-        pass
-
+        for item in other:
+            self.append(item)
+        return self
+        
     def __eq__(self, other):
-        pass
+        return list(self) == other
+        
+    def __ne__(self, other):
+        return list(self) == other
 
     def append(self, elem):
-        pass
+        new_node = Node(elem)
+        if not self.start and not self.end:
+            self.start = new_node
+        else:
+            self.end.next = new_node
+        self.end = new_node
 
     def count(self):
-        pass
+        return len(self)
 
     def pop(self, index=None):
-        pass
+        
+        if index is None:
+            index = self.count() - 1
+            print(index)
+
+        if index > self.__len__() -1:     # if index larger than size
+            raise IndexError
+        
+        if self.start is None:      # if empty list
+            raise IndexError
+        
+        if self.start.next is None: # if one element, set start and end to None
+            temp = self.start
```

Comment:
I would consider using something other than `temp` as the variable name even if it's pretty unambiguous what's going on here.